source/extensions/filters/network/kafka/response_codec.cc

```diff
@@ -0,0 +1,114 @@
+#include "extensions/filters/network/kafka/response_codec.h"
+
+#include "common/common/stack_array.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Kafka {
+
+void ResponseInitialParserFactory::expectResponse(const int16_t api_key,
+                                                  const int16_t api_version) {
+  std::lock_guard<std::mutex> lock{expected_responses_mutex_};
```

Comment:
Why do we need locks in this class?

Reply:
I was unsure how many threads are operating on a filter, so I just took the safe approach :)
A filter needs to have a request codec to insert expected values into response codec (thru this call).
Is it the same thread that handles Filter's `onWrite` and `onData` calls? If yes, then I can remove that lock.

Reply:
Yeah everything is siloed in Envoy, so no locks should be needed.